pkg/internal/api/common.go

```diff
@@ -133,3 +136,28 @@ func (c *ClientImpl) doRequest(ctx context.Context, req *http.Request) ([]byte,
 
 	return body, err
 }
+
+func (c *ClientImpl) runQuery(ctx context.Context, serviceID string, sql string, args ...interface{}) ([]byte, error) {
+	qry, err := sqlbuilder.ClickHouse.Interpolate(sql, args)
+	if err != nil {
+		return nil, err
+	}
+
+	s := struct {
+		Sql string `json:"sql"`
+	}{
+		Sql: qry,
+	}
+
+	rb, err := json.Marshal(s)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(http.MethodPost, c.getQueryAPIPath(serviceID), strings.NewReader(string(rb)))
```

Comment:
```suggestion
	buffer := &bytes.Buffer{}
	if err := json.NewEncoder(buffer).Encode(&s); err != nil {
		return nil, fmt.Errorf("encoding query payload to JSON failed: %w", err)
	}
	req, err := http.NewRequest(http.MethodPost, c.getQueryAPIPath(serviceID), buffer)
```

nit: more straightforward to operate with io.Reader impl